src/v/datalake/coordinator/iceberg_file_committer.cc

```diff
@@ -305,11 +307,17 @@ class table_commit_builder {
           "Adding {} files to Iceberg table {}",
           icb_files_.size(),
           table_id_);
+        // NOTE: a tag is added to the new snapshot to ensure that snapshot
+        // expiry doesn't clear the snapshot with these commit metadata
+        // properties. Tagged snapshots are by default never expired, ensuring
+        // that we retain them e.g. in case of external table updates and low
+        // snapshot retention.
         iceberg::transaction txn(std::move(table_));
         auto icb_append_res = co_await txn.merge_append(
           io,
           std::move(icb_files_),
-          {{commit_meta_prop, to_json_str(commit_meta)}});
+          {{commit_meta_prop, to_json_str(commit_meta)}},
+          commit_tag_name);
```

Comment:
q: maybe useful to document the semantics of overwriting with the same tag name always, what happens to previous tags?